woocommerce/woocommerce
# [Experimental] Product filters/add on sale

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/51855

This is a draft PR to get an idea if this is the direction we want to go. Excuse some of the convoluted code, it will be cleaned up once we agree this is working as expected. Note that I did not implement every single detail as the issue did not call for that. Here are some notes:

- I've added an universal URL parameter called `filter_status` so it would look like `https://site.com/?filter_status=instock,onsale`.
- I've added editor settings `onsale,instock,onbackorder,outofstock`.
- The logic for querying these statues are added to the filter block and to product collection.
- I had to make some changes to the product collection api so I am not sure if we can do that or do we need to iterate a new version.
- I am seeing some similar queries throughout QueryFilters.php and ProductQuery.php and I am not sure why and if we can consolidate those. I will defer this to someone who has worked with these queries for direction.

Things I did not implement and will need follow up PRs:

- Featured status and logic.
- How the filters work when PC and filters are placed in a post or a page.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).
> - Make sure to reset Product Catalog Template.

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template.
2. Edit the template.
3. With the block inserter, add Product Filters (Experimental) block to the template.
4. Highlight the Status Filter Block and go to the Inspector Settings->Settings.
5. You should see the Status panel with all the statuses as checkboxes.
6. Go ahead and enable/disable them and save. 
7. Go to the frontend shop page and ensure you see the enabled statuses shown in the filters.
8. Click on them and ensure they're filtering correctly for the Product Collection.
9. Try enabling counts. Do the same test again and you should now see product counts next to the statuses.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Add on sale, featured status to status block
</details>
